packages/@uppy/companion/src/server/provider/zoom/index.js

```diff
@@ -188,16 +188,18 @@ class Zoom extends Provider {
     let start = end.clone().date(1)
 
     const accountCreation = adapter.getAccountCreationDate(body)
-    const defaultLimit = end.clone().subtract(DEFAULT_RANGE_MOS, 'months')
-    const limit = accountCreation > defaultLimit ? accountCreation : defaultLimit
+    const defaultLimit = end.clone().subtract(DEFAULT_RANGE_MOS, 'months').date(1)
+    const allResultsShown = accountCreation > defaultLimit
+    const limit = allResultsShown ? accountCreation : defaultLimit
 
     const data = {
       items: [],
       username: adapter.getUserEmail(body)
     }
 
-    while (start > limit) {
-      start = end.clone().date(1)
+    while (start.isSameOrAfter(limit, 'month')) {
+      start = end.isSame(limit, 'month') ? limit.clone() : end.clone().date(1)
```

Comment:
maybe we should add a comment here, hinting at what date scenarios this ternary is catching? It took me some time to figure what it was about.